profiler/profiler.go

```diff
@@ -189,6 +195,30 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 	}
 }
 
+// enabledProfileTypes returns the enabled profile types in a deterministic
+// order. The CPU profile always comes first because people might spot
+// interesting events in there and then try to look for the counter-part event
+// in the mutex/heap/block profile. Deterministic ordering is also important
+// for delta profiles, otherwise they'd cover varying profiling periods.
+func (p *profiler) enabledProfileTypes() []ProfileType {
+	order := []ProfileType{
+		CPUProfile,
+		HeapProfile,
+		BlockProfile,
+		MutexProfile,
+		GoroutineProfile,
+		expGoroutineWaitProfile,
+		MetricsProfile,
+	}
+	enabled := []ProfileType{}
+	for _, t := range order {
+		if _, ok := p.cfg.types[t]; ok {
```

Comment:
...Or, if `len(p.cfg.types) != len(enabled)`, I think that's also an error.

Reply:
> ...or, if len(p.cfg.types) != len(enabled), I think that's also an error.

Yeah, I could add a `panic()` for that below the loop. What ya think?

Reply:
Either this or the test case. The test case is probably less brutal.

Reply:
3fcf217